flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/AsyncSinkWriter.java

```diff
@@ -465,7 +489,7 @@ private void addEntryToBuffer(RequestEntryT entry, boolean insertAtHead) {
      * <p>To this end, all in-flight requests need to completed before proceeding with the commit.
      */
     @Override
-    public void flush(boolean flush) {
+    public void flush(boolean flush) throws InterruptedException {
         while (inFlightRequestsCount > 0 || (bufferedRequestEntries.size() > 0 && flush)) {
             mailboxExecutor.tryYield();
```

Comment:
I think this should have been also changed to `yield()`, otherwise we have busy waiting?

Reply:
I completely agree. Thank you so much.

Reply:
On small difference I might suggest here is to have the tryYield line replaced with 
```
        while(inFlightRequestsCount>0){
            mailboxExecutor.yield();
        }
```

Only because it is possible for the code to reach here with `inFlightRequestCount=0`. Then we'd be in a real spot of bother ;)

Reply:
Good point. But do you mean adding this `if` condition beside `while` condition? . In that case wouldn't it be better to squash the two to:
```
while (inFlightRequestsCount > 0 && (bufferedRequestEntries.size() > 0 && flush)) {
  mailboxExecutor.yield();
}
```
?